### ISSUE TITLE
Extra blank line on code block

### ISSUE THREAD

**Issue by pickfire** [rust-unofficial/patterns#155]
![image](https://user-images.githubusercontent.com/4687791/103482824-f8325b80-4e1d-11eb-8fde-acd3cda90dba.png)

https://rust-unofficial.github.io/patterns/functional/index.html

The extra blank line look weird.

**Comment by simonsan** [comment]
I don't see much configuration options for that case under [output.html.playground](https://rust-lang.github.io/mdBook/format/config.html). Probably possible to influence via Custom.js/Custom.css but not sure if this is the scope of this project. Maybe more an `upstream` issue to get a setting for it and a bit of personal preference.

**Comment by pickfire** [comment]
Yeah, after I create the issue I noticed that it doesn't seemed to be an issue for this repo. I thought that it was somehow changed here, I didn't track the change for mdbook so most likely they changed it somewhere.

But weird, their examples does not have the blank lines.

![image](https://user-images.githubusercontent.com/4687791/103560878-89273680-4ef3-11eb-9c62-276ece31ba7d.png)

https://rust-lang.github.io/mdBook/format/theme/syntax-highlighting.html

**Comment by simonsan** [comment]
Please keep us updated if it's an issue we could fix easily with js/css/theme-based approach. ;-)

**Comment by simonsan** [comment]
![2021-01-05 11_24_09-Window-Screenshot](https://user-images.githubusercontent.com/14062932/103635283-96224500-4f48-11eb-9e87-15d743b14fc8.png)

Fixed. Don't asked me why. :-) Feel free to reopen if it isn't fixed on your side still.

**Comment by simonsan** [comment]
mem-replace

![2021-01-05 21_41_39-mem__replace(_) - Rust Design Patterns – Mozilla Firefox-Screenshot](https://user-images.githubusercontent.com/14062932/103696552-d6acad80-4f9e-11eb-8968-1bf295f557bf.png)

dtor-finally
![2021-01-05 21_42_05-Finalisation in Destructors - Rust Design Patterns – Mozilla Firefox-Screenshot](https://user-images.githubusercontent.com/14062932/103696580-e3c99c80-4f9e-11eb-88e2-283efcb0e601.png)

on-stack-dyn-dispatch
![2021-01-05 21_42_27-On-Stack Dynamic Dispatch - Rust Design Patterns – Mozilla Firefox-Screenshot](https://user-images.githubusercontent.com/14062932/103696615-f217b880-4f9e-11eb-81de-caf5504b314f.png)

:-D

Reopening.

**Comment by MarcoIeni** [comment]
Should we keep this opened? I mean that's something that we cannot control..

**Comment by simonsan** [comment]
> Should we keep this opened? I mean that's something that we cannot control..

Yeah, I would keep it open as an `upstream` issue to track. Seeing three different variants in the book I'm also interested now what's going on there, so I would prefer to leave it open.

**Comment by simonsan** [comment]
Tracking: https://github.com/rust-lang/mdBook/issues/1433

**Comment by pickfire** [comment]
I think in some cases it puts an empty space but in some cases it does not, so weird.

I believe adding `# fn main()` there is no new line.

**Comment by MarcoIeni** [comment]
Ok, but I prefer to wait for the actual fix instead of adding `# fn main()` all over the places.
